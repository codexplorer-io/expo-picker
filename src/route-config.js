import { TransitionPresets } from '@react-navigation/stack';
import { Platform } from 'react-native';
import { PICKER_SCREEN_ROUTE_NAME } from './picker';
import { PickerScreen } from './picker-screen';

export const getRouteConfig = () => [{
    name: PICKER_SCREEN_ROUTE_NAME,
    screen: PickerScreen,
    screenOptions: {
        ...Platform.select({
            ios: TransitionPresets.BottomSheetAndroid,
            default: TransitionPresets.ModalSlideFromBottomIOS
        })
    }
}];
