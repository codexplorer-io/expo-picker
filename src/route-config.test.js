import { getRouteConfig } from './route-config';

jest.mock('@react-navigation/stack', () => ({
    TransitionPresets: {
        ModalSlideFromBottomIOS: { transitionPreset: 'ModalSlideFromBottomIOS' }
    }
}));

jest.mock('./picker', () => ({
    PICKER_SCREEN_ROUTE_NAME: 'MockPickerScreenRouteName'
}));

jest.mock('./picker-screen', () => ({
    PickerScreen: 'MockPickerScreen'
}));

describe('getRouteConfig', () => {
    it('should return expected value', () => {
        expect(getRouteConfig()).toMatchSnapshot();
    });
});
