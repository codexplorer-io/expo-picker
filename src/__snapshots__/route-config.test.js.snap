// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRouteConfig should return expected value 1`] = `
Array [
  Object {
    "name": "MockPickerScreenRouteName",
    "screen": "MockPickerScreen",
    "screenOptions": Object {
      "transitionPreset": "ModalSlideFromBottomIOS",
    },
  },
]
`;
